### PR TITLE
Fix e2e test

### PR DESCRIPTION
**Pull request by YangKeao**
### What problem does this PR solve?

It fixes e2e test. There are two problems in the e2e test now:

1. #1777 breaks the e2e script, as shown in #1815
2. The IOChaos e2e test failed message check was wrong

### CODE REVIEW

**Review by STRRL — APPROVED** [review]

File: hack/run-e2e.sh [hunk 1]
```diff
@@ -177,4 +177,4 @@ if [ -n "$REPORT_DIR" ]; then
 fi
 
 echo "info: docker ${docker_args[@]} $E2E_IMAGE ${e2e_args[@]}"
-docker "${docker_args[@]}" $E2E_IMAGE "${e2e_args[@]}"
+docker ${docker_args[@]} $E2E_IMAGE ${e2e_args[@]}
```
[the remark this answers is not in the document]
Comment (cwen0) [reply hack/run-e2e.sh:180]: @muse-dev ignore

File: hack/local-up-chaos-mesh.sh [hunk 2]
```diff
@@ -117,7 +117,7 @@ images=(
     $DOCKER_REGISTRY_PREFIX/pingcap/chaos-dashboard:${IMAGE_TAG}
     $DOCKER_REGISTRY_PREFIX/pingcap/chaos-daemon:${IMAGE_TAG}
 )
-for n in "${images[@]}"; do
+for n in ${images[@]}; do
```
[the remark this answers is not in the document]
Comment (cwen0) [reply hack/local-up-chaos-mesh.sh:120]: @Muse-Dev ignore

File: hack/e2e.sh [hunk 3]
```diff
@@ -333,4 +333,4 @@ if [ -n "${ARTIFACTS}" -a -z "$SKIP_DUMP" ]; then
 fi
 
 echo "info: run 'kubetest2 ${kubetest2_args[@]} -- hack/run-e2e.sh $@'"
-$KUBETSTS2_BIN "${kubetest2_args[@]}" -- hack/run-e2e.sh "$@"
+$KUBETSTS2_BIN ${kubetest2_args[@]} -- hack/run-e2e.sh "$@"
```
[the remark this answers is not in the document]
Comment (cwen0) [reply hack/e2e.sh:336]: @Muse-Dev ignore

**Pull request merged**